hpy/devel/src/runtime/ctx_type.c

```diff
@@ -822,7 +906,7 @@ ctx_Type_FromSpec(HPyContext *ctx, HPyType_Spec *hpyspec,
     spec->basicsize = basicsize;
     spec->flags = flags | HPy_TPFLAGS_INTERNAL_IS_HPY_TYPE;
     spec->itemsize = hpyspec->itemsize;
-    spec->slots = create_slot_defs(hpyspec, base_member_offset, extra);
+    spec->slots = create_slot_defs(hpyspec, (HPy_ssize_t)base_member_offset, extra);
```

Comment:
same question as above, I think the cast here is because you turned it into an `int`

Reply:
I switched to `int` mostly because `HPyType_Spec.basicsize` also is an `int`. But I can use `HPy_ssize_t` here everywhere. This is, for sure, safer.

Reply:
Ah, no, I now remember:
If we use `HPy_ssize_t` for `basicsize` and `base_member_offset`, we need to cast from `HPy_ssize_t` to `int` at some point which is unsafe. In this way, we just need to cast from `int` to `HPy_ssize_t` which should be safe.

Reply:
> we need to cast from HPy_ssize_t to int at some point which is unsafe. 

I guess that this "some point' is inside the CPython implementation of HPy, i.e. when you need to fill CPython's `PyType_Spec.basicsize`, is it correct?

If so, I don't think it is unsaf(er) than what we have now.
In C, the return type of `offsetof` and `sizeof` is `size_t`, so potentially they might not fit into an `int` (veeeeery potentially, I admit 😅). So, in one case the "unsafe cast" between `size_t` and `int` is done by the user, in the other it's done by us to comply with a (broken, IMHO) CPython API.

So, I think I would prefer to do the "clean & correct" thing on the user-facing side and to the "unsafe necessary cast" as an implementation detail (and possibly also throw `HPy_FatalError` in case it doesn't fit).

That said, I don't think it changes anything in practice at the moment of course (it works on CPython, so it's battle-tested by thousands of extensions and many compilers)

Reply:
Okay, you convinced me. I switched back to `HPy_ssize_t`.